### PR TITLE
feat: centrifuge base

### PR DESCRIPTION
**Pull request by bcai2**
- Adds the `centrifuge` tool.
- Adds docs.
- Refactors how `prefix_mapping` is generated for `megahit` with a new module (`input_util.py`) and function (`convert_input_params_to_prefix_mapping`). Adds a unit test for the function.

### CODE REVIEW

File: toolchest_client/tools/tool_args.py [hunk 1]
```diff
@@ -206,6 +206,29 @@
...
+            "--al-conc",
+            "--un-gz",
+            "--un-bz2",
+            "--al-gz",
+            "--al-bz2",
+            "--un-conc-gz",
+            "--un-conc-bz2",
+            "--al-conc-gz",
+            "--al-conc-bz2",
+        ]
```
Comment (lebovic) [inline toolchest_client/tools/tool_args.py:230]: Does this config actually reject non-blacklisted args that aren't the first item in `tool_args`?

File: toolchest_client/files/input_util.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+"""
+toolchest_client.files.input_util
```
Comment (lebovic) [inline toolchest_client/files/input_util.py:2]: I agree that this is a good point to separate this logic into it's own module.

The name (`input_util`) and logical grouping ("Utilities for handling input files/paths") doesn't make sense to me, though. What functions do we put in here, but not `general` or another module that pops up for input files or output files?

Would this fit in `general`, or could this be renamed?

**Review by lebovic — APPROVED** [review]

File: toolchest_client/files/input_util.py (hunk 2 above)
Comment (bcai2) [reply toolchest_client/files/input_util.py:2]: Done re: the unit test! It's in `toolchest_client/files/tests/test_input_params.py`.
I agree that the name isn't great. I'd like to refactor it, but I'm still trying to think of a logical name.

File: toolchest_client/tools/tool_args.py (hunk 1 above)
Comment (bcai2) [reply toolchest_client/tools/tool_args.py:230]: I'm not sure what you mean by this, can you give me an example config? (I don't believe so if I'm understanding correctly, but I just want to make sure that I am.)

File: toolchest_client/files/input_util.py (hunk 2 above)
Comment (bcai2) [reply toolchest_client/files/input_util.py:2]: I'm going to refactor this into `general.py` and move on.

**Comment by bcai2** [comment]
@lebovic I'm merging this, but I also left replies and made a refactor that would be good to look over.

**Pull request merged**